.travis.yml

```diff
@@ -0,0 +1,35 @@
+# Config file for automatic testing at travis-ci.org
+sudo: true 
+branches:
+  only:
+    - master
+    - test
+language: python
+python:
+  - 3.6
+services:
+  - postgresql
+install: 
+  - pip install -r requirements.txt
+before_script:
+  - psql -c 'create database ether_sql;' -U postgres
+  - psql -c "CREATE USER root WITH PASSWORD 'develop';" -U postgres
+  - psql ether_sql < tests/ether_sql_test.sql
+  - export FLASK_APP="api/api.py"
+  - export FLASK_DEBUG=1
+
+script:
+  - |
+    python api/api.py &
+    APP_PID=$!
+  - curl -s https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-linux-amd64.zip > ngrok.zip
+  - unzip ngrok.zip
+  - ./ngrok http 5000 > /dev/null &
+
+  # sleep to allow ngrok to initialize
+  - sleep 2
+
+  # extract the ngrok url
+  - NGROK_URL=$(curl -s localhost:4040/api/tunnels/command_line | jq --raw-output .public_url)
+  - curl -i $NGROK_URL/v1.0/current_blockNumber/
```

Comment:
I would prefer that we use the pytest to assert that the above function passes eg: http://flask.pocoo.org/docs/1.0/testing/#the-first-test

Reply:
http://flask.pocoo.org/docs/1.0/testing/#testing-json-apis

Reply:
Which function? The one that gets the ngrok_url?

Reply:
Do you mean to say that we call each of those api functions in a pytest function and then call that pytest function in Travis, instead of using curl for each individual api link? 
What's wrong with using curl? It ensures that the api is handling requests properly, and if even any one API link doesn't work the test won't pass, isn't that what we want?

Reply:
I prefer pytest mostly because it is very easy to write and deploy tests.
Pytest enables us to test individual python functions, you can also use the curl functions by using something like: https://docs.python.org/2/library/subprocess.html#using-the-subprocess-module

Reply:
So you mean in Travis after we have installed and built stuff, we should run a pytest module first to test individual function, and then run the server and check the curl requests too like we are doing right now, so that it covers everything?